plugins/destination/file/client/write.go

```diff
@@ -17,6 +18,11 @@ func (c *Client) WriteTableBatch(ctx context.Context, arrowSchema *arrow.Schema,
 	tableName := schema.TableName(arrowSchema)
 	timeNow := time.Now().UTC()
 	p := replacePathVariables(c.pluginSpec.Path, tableName, c.pluginSpec.Format, uuid.NewString(), timeNow)
+
+	if err := os.MkdirAll(path.Dir(p), 0755); err != nil {
```

Comment:
This call was missing in the new 'path scheme', but it was creating directories with template variables in them in `client.New` (which was removed, above)